application/comit_node/src/swap_protocols/rfc003/secret.rs

```diff
@@ -94,24 +128,26 @@ impl From<Secret> for SecretHash {
 }
 
 impl Secret {
-    pub const SECRET_LENGTH: usize = 32;
+    // Both values need to stay the same!
+    pub const LENGTH: usize = 32;
+    pub const LENGTH_U8: u8 = 32;
 
     pub fn generate<T: RandomnessSource>(rng: &mut T) -> Secret {
-        let random_bytes = rng.gen_random_bytes(Self::SECRET_LENGTH);
-        let mut secret = [0; Self::SECRET_LENGTH];
+        let random_bytes = rng.gen_random_bytes(Self::LENGTH);
+        let mut secret = [0; Self::LENGTH];
         secret.copy_from_slice(&random_bytes[..]);
         Secret::from(secret)
     }
 
-    pub fn from_vec(vec: &[u8]) -> Result<Secret, SecretFromErr> {
-        if vec.len() != Self::SECRET_LENGTH {
-            return Err(SecretFromErr::InvalidLength {
-                expected: Self::SECRET_LENGTH,
+    pub fn from_vec(vec: &[u8]) -> Result<Secret, FromErr> {
```

Comment:
❓Same as above: Can't you make `vec: &[u8]` already a fixe length array and get rid of the check below?

Reply:
No can do for `Secret` as we will need to extract it from a Bitcoin Script that hold `Vec<u8>`